voter.py

```diff
@@ -0,0 +1,67 @@
+from ballot import Ballot, Vote
+from utils import rsa
+
+import json
+import secrets
+
+class Voter:
+
+    def __init__(self, ident: int, keypair: tuple, admin_key: tuple):
+        self.ident = ident
+        self.N, self.publicExp, self.privateExp = keypair
+        self.admin_key = admin_key # (N, public, private)
+
+    def cast(self, vote_str: str):
+        """Commit and sign the voter's vote, returning a tuple of (identity, committed vote, signature).
+        This gets sent to the administrator for authorization in the form of a blind signature.
+        """
+        # Read in vote JSON and attach random identifier to vote so we can identify it later
+        vote = json.loads(vote_str)
+        self.vote_uuid = secrets.randbits(64)
+        vote["uuid"] = self.vote_uuid
+        vote_str = json.dumps(vote)
+        vote_bytes = bytes(vote_str.encode('utf-8'))
+
+        # Create random commitment key the same length as the vote
+        commitment_key_int = secrets.randbits(8 * len(vote_bytes))
+        self.commitment_key = commitment_key_int.to_bytes(len(vote_bytes), 'big')
+        self.committed_vote = rsa.xor(vote_bytes, self.commitment_key)
+        blinded_vote = rsa.blind_fdh(self.committed_vote, self.admin_key[1], self.admin_key[0])
+        signature = rsa.sign_fdh(self.committed_vote, self.privateExp, self.N)
```

Comment:
What's going to use this signature? 

Administrator looks like it verifies the signature on the blinded vote rather tha this commited vote.

Reply:
Ah shoot, meant to put `self.blinded_vote`, good catch